sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/TableCatalog.java

```diff
@@ -47,6 +47,12 @@ public interface TableCatalog extends CatalogPlugin {
    */
   String PROP_LOCATION = "location";
 
+  /**
+   * A reserved property to indicate that the table location is managed, not user-specified.
+   * If this property is "true", SHOW CREATE TABLE will not generate the LOCATION clause.
+   */
+  String PROP_IS_MANAGED_LOCATION = "is_managed_location";
```

Comment:
nit :
actually, we can add a reserved property named `PROP_TABLE_TYPE`, because type include `EXTERNAL` `MANAGED` `VIEW` that also can control the different behaviors when `TYPE == MANAGED`. And  `PROP_TABLE_TYPE` sounds more generic, may be we can add more type in future.

Reply:
I don't think we will use `TableCatalog` to support views. I'm adding this new property as I think this is the most precise way. People can create EXTERNAL table without location, or create MANAGED TABLE with location. What we care in SHOW CREATE TABLE is if the location is managed or not, instead of the table type.